woocommerce-gateway-ebanx/services/class-wc-ebanx-query-router.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+if (!defined('ABSPATH')) {
+	exit;
+}
+
+class WC_EBANX_Query_Router {
+	private $key = null;
+	private $routes = array();
+
+	/**
+	 * Construct the router using a query string key
+	 *
+	 * @param string $key
+	 */
+	public function __construct($key) {
+		self::validate_key($key);
+		$this->key = $key;
+	}
+
+	/**
+	 * Maps a key value to a handler method
+	 *
+	 * @param  string $key_value The value which triggers this route
+	 * @param  array  $handler   A callable array with instance/class and method
+	 * @return void
+	 */
+	public function map($key_value, $handler) {
+		self::validate_key_value($key_value);
+		self::validate_route($handler);
+
+		$this->routes[$key_value] = $handler;
+	}
+
+	/**
+	 * The dispatcher that will make coffee for you if you ask nicely
+	 * Just call this at the end of your setup
+	 *
+	 * @return void
+	 */
+	public function serve() {
+		$route = $this->find_route_for_key();
+		if ($route === null) {
+			// Not found, carry on
+			return;
+		}
+
+		$params = self::read_request_for_route($route);
+
+		// Route it and die
+		call_user_func_array($route, $params);
+		die();
+	}
+
+	// Private
+
+	/**
+	 * Locates the desired route based on the key value
+	 *
+	 * @param  string $search_key A key to read the value from (optional)
+	 * @return array              A callable array of your route handler
+	 */
+	private function find_route_for_key($search_key = null) {
+		$key = $this->key;
+		if ($search_key !== null) {
+			$key = $search_key;
+		}
+
+		self::validate_key($key);
+
+		if (!WC_EBANX_Request::has($key)) {
+			// Not set, not found
+			return null;
+		}
+
+		$value = WC_EBANX_Request::read($key);
+
+		self::validate_key_value($value);
+
+		// Not found
+		if (!isset($this->routes[$value])) {
```

Comment:
Hmmm... I think it would be more readable if you use `if ( ! array_key_exists($value) )`.
Also, don't forget we are using spaces after every `(` or `!` and before every `)`